sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -0,0 +1,573 @@
+        return wheel_filename
+      else:
+        raise RuntimeError('Unrecognized SDK wheel file: %s' % sdk_location)
+    else:
+      return names.DATAFLOW_SDK_TARBALL_FILE
+
+  def _stage_beam_sdk(self, sdk_remote_location, staging_location, temp_dir):
+    """Stages a Beam SDK file with the appropriate version.
+
+      Args:
+        sdk_remote_location: A URL from which thefile can be downloaded or a
+          remote file location. The SDK file can be a tarball or a wheel. Set
+          to 'pypi' to download and stage a wheel and source SDK from PyPi.
+        staging_location: Location where the SDK file should be copied.
+        temp_dir: path to temporary location where the file should be
+          downloaded.
+
+      Returns:
+        A list of SDK files that were staged to the staging location.
+
+      Raises:
+        RuntimeError: if staging was not successful.
+      """
+    if sdk_remote_location == 'pypi':
+      sdk_local_file = self._download_pypi_sdk_package(temp_dir)
+      sdk_sources_staged_name = self._desired_sdk_filename_in_staging_location(
+          sdk_local_file)
+      staged_path = FileSystems.join(staging_location, sdk_sources_staged_name)
+      logging.info('Staging SDK sources from PyPI to %s', staged_path)
+      self.stage_artifact(sdk_local_file, staged_path)
+      staged_sdk_files = [sdk_sources_staged_name]
+      try:
+        # Stage binary distribution of the SDK, for now on a best-effort basis.
+        sdk_local_file = self._download_pypi_sdk_package(
+            temp_dir, fetch_binary=True)
+        sdk_binary_staged_name = self._desired_sdk_filename_in_staging_location(
+            sdk_local_file)
+        staged_path = FileSystems.join(staging_location, sdk_binary_staged_name)
+        logging.info('Staging binary distribution of the SDK from PyPI to %s',
+                     staged_path)
+        self.stage_artifact(sdk_local_file, staged_path)
+        staged_sdk_files.append(sdk_binary_staged_name)
+      except RuntimeError as e:
+        logging.warn(
+            'Failed to download requested binary distribution '
+            'of the SDK: %s', repr(e))
+
+      return staged_sdk_files
+    elif self._is_remote_path(sdk_remote_location):
+      local_download_file = os.path.join(temp_dir, 'beam-sdk.tar.gz')
+      self._download_file(sdk_remote_location, local_download_file)
+      staged_name = self._desired_sdk_filename_in_staging_location(
+          sdk_remote_location)
+      staged_path = FileSystems.join(staging_location, staged_name)
+      logging.info('Staging Beam SDK from %s to %s', sdk_remote_location,
+                   staged_path)
+      self.stage_artifact(local_download_file, staged_path)
+      return [staged_name]
+    else:
+      raise RuntimeError(
+          'The --sdk_location option was used with an unsupported '
+          'type of location: %s' % sdk_remote_location)
+
+  def _download_pypi_sdk_package(self,
```

Comment:
<!--new_thread; commit:fb22e20e257e1c00a618286cb0a1dd01e4071709; resolved:0-->
Make this a private function in Stager module.

Reply:
I would like to keep all the functionality of Stager within the class instead of fragmenting it between Stager class and module. 
As none of these functions are reusable and are not intended to be reused in anyway, splitting things in module and class will only make code harder to follow and maintain.